### PR TITLE
Bare metal: set up basic CPU exception handling, with IDT & TSS

### PR DESCRIPTION
**Pull request by tkchia**
When a CPU exception (e.g. protection fault, divide by zero) happens in bare metal mode, the program should now be able to catch the exception and dump some information on the exception to the serial console, if one ropes in (`STATIC_YOINK`) the symbol `_idt`.

I hope to get the exception handlers to dump information to the VGA screen as well.  It seems I will need to modify `kprintf` to do this.  Once this is implemented, it should be easier to diagnose any program crashes that occur on bare metal.

Thank you!

### CODE REVIEW

**Comment by ghaerr** [comment]
Amazing work, @tkchia!

File: libc/intrin/interrupts.S [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	mov	%es,%eax
+	push	%rax
+	mov	$GDT_LONG_DATA,%eax	# ...& load ds, es, ss correctly
+	mov	%eax,%ss
+	mov	%eax,%ds
+	mov	%eax,%es
+	ezlea	.excep_msg,di		# stack should be 16-byte aligned now
+	xor	%eax,%eax		# kprintf is variadic; remember to
+					# pass no. of vector regs. used (= 0)
+	call	kprintf			# print error message
```
Comment (jart) [inline libc/intrin/interrupts.S:94]: Is this going to force a dependency on kprintf() into o/tiny/examples/life.com? If so, we might want to do a one-off error report.

File: libc/intrin/interrupts.S [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+	lidt	_idtr			# load IDTR
+	ezlea	_idt,di
+	pushpop	32,%rcx			# fill IDT entries for CPU exceptions
+	ezlea	__excep0_isr,dx
+1:	mov	%edx,%eax
+	stosw
+	mov	%cs,%eax
+	stosw
+	mov	%rdx,%rax
+//		   ┌P:present
```
Comment (jart) [inline libc/intrin/interrupts.S:126]: I love how you did the CP437 diagrams.

**Review by jart — APPROVED** [review]
I can't say how impressed I am with this change. Doing this is something that's mystified me for quite some time.

File: libc/intrin/interrupts.S (hunk 1 above)
Comment (tkchia) [reply libc/intrin/interrupts.S:94]: Hello @jart,

> Is this going to force a dependency on kprintf() into o/tiny/examples/life.com?

Not yet.  I am trying to arrange to link in the IDT logic only when the program is already linking in some sort of crash debugging (I am still not familiar with how Cosmopolitan's crash debugging capability works).

Thank you!

**Comment by tkchia** [comment]
Hello @jart,

> Doing this is something that's mystified me for quite some time.

Well, it did take me quite a while to realize that I had to reload the GDTR to make it point to virtual memory &mdash; so that, when (re)loading `%cs` to service an interrupt, the CPU could properly load `%cs` 's descriptor from main memory, by way of the GDTR.  I am not sure if this was the problem you encountered.

Thank you!

**Pull request merged**